src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,132 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int NATURAL_NUMBER_BORDERLINE = 0;
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double CAPACITY_MULTIPLY_BY = 1.5;
+    private static final String NEGATIVE_INDEX_ERROR = "Index should be natural number";
+    private static final String INDEX_OUT_OF_SIZE_ERROR = "Index out of list length";
+
+    private T[] data;
+    private int size;
+
+    public ArrayList() {
+        data = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            data = (T[]) new Object[capacity];
+        } else if (capacity == 0) {
+            data = (T[]) new Object[]{};
+        } else {
+            throw new ArrayListIndexOutOfBoundsException(NEGATIVE_INDEX_ERROR);
+        }
+    }
+
```

Comment:
I'm not sure that you need to create two constructors. I think the second constructor is useless, also if u throw error it should be IllegalArgumentException.

Reply:
I think it should stay because the original ArrayList has it. But the exception type is wrong, agree

Reply:
I'm not saying that it's a mistake, just that it's not necessary because this method is not in the interface and you don't use it anywhere else